internal/certs/certs.go

```diff
@@ -14,15 +14,17 @@ import (
 )
 
 const (
-	bitSize = 4096
-	base    = 2
-	exp     = 159
+	bitSize      = 2048
+	base         = 1
+	exp          = 128
+	caCommonName = "kubewarden-controller-ca"
 )
 
 // GenerateCA generates a self-signed CA root certificate and private key in PEM format.
 // It accepts validity bounds as parameters.
 func GenerateCA(notBefore, notAfter time.Time) ([]byte, []byte, error) {
-	serialNumber, err := rand.Int(rand.Reader, (&big.Int{}).Exp(big.NewInt(base), big.NewInt(exp), nil))
+	serialNumberUpperBound := new(big.Int).Lsh(big.NewInt(base), exp)
```

Comment:
nitpicking: this is not really a `base`. the `Lsh` function does a left shift, which in this case translates to: `1 * 2^128`. Where `1` is the number multiplied by "2 power `exp`".

I would:

* remove the `base` constant and just use `1` inside of the code
* remove the `exp` constant, or rename it to `serial_upper_bound`